### PR TITLE
Update build action

### PR DESCRIPTION
**Pull request by nathane**
- Test Laravel 8
- Update `actions/cache@v2`
- Added `nick-invision/retry@v1` to handle composer timeouts.
- Setup problem matchers for `PHP` & `PHPUnit`

    >This will scan the logs for PHP & PHPUnit errors and warnings, and surface them prominently in the GitHub Actions UI by creating annotations and log file decorations.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -58,12 +58,16 @@ jobs:
...
+
       - name: Cache dependencies
-        uses: actions/cache@v1
+        uses: actions/cache@v2
         with:
-          path: ~/.composer/cache/files
+          path: ${{ steps.determine-composer-cache-directory.outputs.directory }}
           key: dependencies-os-${{ matrix.os }}-php-${{ matrix.php-version }}-laravel-${{ matrix.laravel-version }}-composer-${{ hashFiles('**/composer.lock') }}
-          restore-keys: dependencies-os-${{ matrix.os }}-php-${{ matrix.php-version }}-laravel-${{ matrix.laravel-version }}
+          restore-keys: dependencies-os-${{ matrix.os }}-php-${{ matrix.php-version }}-laravel-${{ matrix.laravel-version }}-composer-
```
Comment (jasonmccreary) [inline .github/workflows/build.yml:70]: @nathane, could you explain the change to this line?

Comment (nathane) [reply .github/workflows/build.yml:70]: Speed up the cache restoring process by partially matching the `key`. `restore-keys` used for finding the cache if no cache hit occurred for `key`.

> - When there is an exact match, the action restores the files in the cache to the path directory.
> - If there are no exact matches, the action searches for partial matches of the restore keys. When the action finds a partial match, the most recent cache is restored to the path directory.
https://docs.github.com/en/actions/configuring-and-managing-workflows/caching-dependencies-to-speed-up-workflows

Comment (jasonmccreary) [reply .github/workflows/build.yml:70]: I see. So technically this was working before, you just added the final `-composer-` to the key.

Comment (nathane) [reply .github/workflows/build.yml:70]: yes, that is correct.

**Pull request merged**